packages/client/src/lib.js

```diff
@@ -99,9 +100,18 @@ class NFTStorage {
    * @returns {Promise<CIDString>}
    */
   static async storeBlob(service, blob) {
-    const { cid, car } = await NFTStorage.encodeBlob(blob)
-    await NFTStorage.storeCar(service, car)
-    return cid.toString()
+    const blockstore = new Blockstore()
+    let cidString
+
+    try {
+      const { cid, car } = await NFTStorage.encodeBlob(blob, blockstore)
+      await NFTStorage.storeCar(service, car)
```

Comment:
We can as an alternative do this within storeCar, but we would need to provide blockstore here, which feels odd as we don't even use `ipfs-car` related things in the function. Also, wrapping everything to guarantee error handling looked the best option

Reply:
In my opinion, this can be redesigned as if a user want to store a big chunk, let's say 10GB, they need to have it available in tmpfs at once. I would recommend to have only the last pending blocks in the tmpfs and delete them after uploading.

Reply:
We are working on a new solution to pack files that will stream directly instead of using a temporary blockstore to keep the blocks. So, this will be an enhancement that we will ship soon to improve this behaviour https://github.com/nftstorage/nft.storage/issues/980